### PR TITLE
Add possibility to display the dial in numbers on the welcome page

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        const { url } = request;
+
+        if (url.startsWith('tel:')) {
+            Linking.openURL(url);
+            this.props.dispatch(setDialInSummaryVisible());
+
+            return false;
+        }
+
+        return true;
```
Comment (virtuacoplenny) [inline react/features/invite/components/dial-in-summary/native/DialInSummary.js]: Do you see a chance that the webview could navigate somewhere unintended, which might also need blocking?

File: react/features/invite/functions.js [hunk 2]
```diff
@@ -389,6 +390,20 @@ export function searchDirectory( // eslint-disable-line max-params
             });
 }
 
+/**
+ * Returns true if the current deployment supports dial number retreival.
+ *
+ * @param {Object | Function} stateful - An instance that can be resolved into the Redux state.
+ * @returns {boolean}
+ */
+export function canGetDialInfo(stateful: Object | Function): boolean {
```
[the remark this answers is not in the document]
Comment (virtuacoplenny) [reply react/features/invite/functions.js]: Although now that I think about it, maybe this should be getting called? It's possible a deployment does not have dial in enabled. How would that case be handled?

**Comment by zbettenbuk** [comment]
Swipe-to-dial

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+     * @returns {void}
+     */
+    _onCloseView() {
+        this.props.dispatch(setDialInSummaryVisible());
+    }
+
+    _onNavigate: Object => Boolean;
+
+    /**
+     * Callback to intercept navigation inside the webview and make the nastive app handle the dial requests.
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/invite/components/dial-in-summary/native/DialInSummary.js]: That's when native gets nasty -> nastive.

How you don't understand that?

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/invite/components/dial-in-summary/native/DialInSummary.js]: Good point!

File: react/features/recent-list/components/RecentList.native.js [hunk 4]
```diff
@@ -107,6 +114,18 @@ class RecentList extends AbstractRecentList<Props> {
     _onDelete(itemId) {
         this.props.dispatch(deleteRecentListEntry(itemId));
     }
+
+    _onShowNumbers: Object => void
+
+    /**
+     * Callback for the numbers action of the list.
+     *
+     * @param {Object} itemId - The ID of the entry for which we'd like to show the dial in numbers.
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/recent-list/components/RecentList.native.js]: Not really, but it's still an ID (maybe more like a complex key?), used to identify an item in the recent list, as they don't have generated simple IDs.

File: react/features/invite/functions.js [hunk 5]
```diff
@@ -508,6 +523,22 @@ export function getDialInfoPageURL(
     return `${origin}${newPath}/static/dialInInfo.html?room=${conferenceName}`;
 }
 
+/**
+ * Generates the URL for the static dial in info page.
+ *
+ * @param {string} uri - The conference URI string.
+ * @returns {string}
+ */
+export function getDialInfoPageURLForURIString(
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/invite/functions.js]: Not really, I just wanted to provide another way to generate this URL. I took the comment above the ```getDialInfoPageURL``` function to decide to go with another implementation: ```The current location URL, the object coming from state ['features/base/connection'].locationURL``` as we don't have a connection at this point yet.

File: android/settings.gradle [hunk 6]
```diff
@@ -21,3 +21,5 @@ include ':react-native-webrtc'
 project(':react-native-webrtc').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-webrtc/android')
 include ':react-native-calendar-events'
 project(':react-native-calendar-events').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-calendar-events/android')
+include ':react-native-webview'
```
Comment (saghul) [inline android/settings.gradle]: can you please sort it properly? looks like it was sorted ok until calendar-events was added

File: ios/Podfile [hunk 7]
```diff
@@ -54,6 +54,8 @@ target 'JitsiMeet' do
   pod 'RNWatch', :path => '../node_modules/react-native-watch-connectivity'
   pod 'react-native-calendar-events',
     :path => '../node_modules/react-native-calendar-events'
+  pod 'react-native-webview',
```
Comment (saghul) [inline ios/Podfile]: ditto, let's sort this please

File: lang/main.json [hunk 8]
```diff
@@ -773,6 +773,7 @@
         "enterRoomTitle": "Start a new meeting",
         "go": "GO",
         "join": "JOIN",
+        "numbers": "Numbers",
```
Comment (saghul) [inline lang/main.json]: "Info" ?

File: react/features/invite/actionTypes.js [hunk 9]
```diff
@@ -41,6 +41,16 @@ export const REMOVE_PENDING_INVITE_REQUESTS
...
 
+/**
+ * The type of Redux action to set the visibility of the dial in summary.
+ *
+ * {
+ *     type: SET_DIAL_IN_SUMMARY_VISIBLE,
+ *     visible: boolean
+ * }
+ */
+export const SET_DIAL_IN_SUMMARY_VISIBLE = 'SET_DIAL_IN_SUMMARY_VISIBLE';
```
Comment (saghul) [inline react/features/invite/actionTypes.js]: (Not necessarily applying to this PR but still). This functionality is not really related to inviting others to the meeting. Maybe we need a new one.

File: react/features/invite/actions.js [hunk 10]
```diff
@@ -268,3 +269,19 @@ export function removePendingInviteRequests() {
...
     };
 }
+
+/**
+ * Action to set the visibility of the dial in summary.
+ *
+ * @param {?string} locationUrl - The location URL to show the dial in summary for.
+ * @returns {{
+ *     type: SET_DIAL_IN_SUMMARY_VISIBLE,
+ *     visible: boolean
```
Comment (saghul) [inline react/features/invite/actions.js]: doesn't match the implementation

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
+// @flow
+
+import React, { Component } from 'react';
+import { type Dispatch } from 'redux';
```
Comment (saghul) [inline react/features/invite/components/dial-in-summary/native/DialInSummary.js]: style: alpha sort by package name

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+     * @inheritdoc
+     */
+    render() {
+        const { _summaryUrl } = this.props;
+
+        return (
+            <SlidingView
+                position = 'bottom'
+                show = { Boolean(_summaryUrl) } >
+                <View style = { styles.webViewWrapper }>
```
Comment (saghul) [inline react/features/invite/components/dial-in-summary/native/DialInSummary.js]: Don't we have this thing (header with back button and label + content) in about 3 places now? How about having some common entity?

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+                    <Header>
+                        <BackButton onPress = { this._onCloseView } />
+                        <HeaderLabel labelKey = 'info.label' />
+                    </Header>
+                    <WebView
```
Comment (saghul) [inline react/features/invite/components/dial-in-summary/native/DialInSummary.js]: We should handle onError here: https://github.com/react-native-community/react-native-webview/blob/master/docs/Reference.md#onerror

Comment (saghul) [reply react/features/invite/components/dial-in-summary/native/DialInSummary.js]: Also, what are we showing while it loads? Maybe we should have a spinner covering it

File: react/features/invite/functions.js [hunk 14]
```diff
@@ -389,6 +390,20 @@ export function searchDirectory( // eslint-disable-line max-params
...
+/**
+ * Returns true if the current deployment supports dial number retreival.
+ *
+ * @param {Object | Function} stateful - An instance that can be resolved into the Redux state.
+ * @returns {boolean}
+ */
+export function canGetDialInfo(stateful: Object | Function): boolean {
+    const { dialInConfCodeUrl, dialInNumbersUrl, hosts }
+                = toState(stateful)['features/base/config'];
+    const mucURL = hosts && hosts.muc;
```
Comment (saghul) [inline react/features/invite/functions.js]: Not sure if checking the MUC URL is too much, without it nothing will work.

File: react/features/invite/functions.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/invite/functions.js]: I think this function should take a location URL and a room instead. If / when when change the URL parsing code to be saner, we'll have one less place to chase.

File: react/features/recent-list/components/RecentList.native.js [hunk 15]
```diff
@@ -79,6 +82,10 @@ class RecentList extends AbstractRecentList<Props> {
         } = this.props;
         const recentList = toDisplayableList(_recentList, t, _defaultServerURL);
         const slideActions = [ {
+            backgroundColor: ColorPalette.blue,
+            onPress: this._onShowNumbers,
```
Comment (saghul) [inline react/features/recent-list/components/RecentList.native.js]: info

File: react/features/invite/functions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/invite/functions.js]: No, this is not invoked, it's a remainder of a previous implementation. Thanks!

File: css/modals/invite/_info.scss [hunk 16]
```diff
@@ -185,6 +185,7 @@
     font-size: 12px;
     max-height: 100%;
     overflow: auto;
+    padding: 15pt;
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply css/modals/invite/_info.scss]: On mobile we prefer pt, because it's relative to the dpi setting of the screen and works generally better. This setting will not affect browsers, as they have a much bigger screen than a padding would kick in.

File: react/features/invite/functions.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/invite/functions.js]: But then I need to parse the url to get the room _before_ invoking this function, so I can't really see what's the win

File: react/features/invite/actions.js [hunk 17]
```diff
@@ -268,3 +269,19 @@ export function removePendingInviteRequests() {
...
+ * }}
+ */
+export function setDialInSummaryVisible(locationUrl: ?string) {
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/invite/actions.js]: Yeah, I was thinking about it, it's kind of a mix of visibility control and property setting. Now replaced it with 2 functions instead: ```showDialInSummary``` and ```hideDialInSummary```. This should be a bit easier to understand

File: lang/main.json (hunk 8 above)
Comment (zbettenbuk) [reply lang/main.json]: Tried the 2 lines rendering again, didn't work, so I fell back to "Dial-in". What do you think?

File: react/features/invite/actionTypes.js (hunk 9 above)
Comment (zbettenbuk) [reply react/features/invite/actionTypes.js]: I agree, but I saw that the web implementation of this component was here, so I wanted to stay consistent.

File: react/features/invite/functions.js (hunk 14 above)
Comment (zbettenbuk) [reply react/features/invite/functions.js]: Right, but this is not invoked, it was just a remainder of a refactor so I removed it. Thanks!

File: react/features/recent-list/components/RecentList.native.js (hunk 15 above)
Comment (zbettenbuk) [reply react/features/recent-list/components/RecentList.native.js]: Renamed to ```_onShowDialInInfo```. I think that covers the functionality and doesn't depend on the label that we may still change

File: lang/main.json (hunk 8 above)
Comment (saghul) [reply lang/main.json]: Sounds good!

Comment (saghul) [reply lang/main.json]: It may look worse on other languages though.

File: react/features/invite/functions.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/invite/functions.js]: We might need to change how we store the recents list at some point...

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js (hunk 12 above)
Comment (zbettenbuk) [reply react/features/invite/components/dial-in-summary/native/DialInSummary.js]: Right! I extract it to a component, thanks!

File: lang/main.json (hunk 8 above)
Comment (zbettenbuk) [reply lang/main.json]: Ok, wen't back to ```Info``` :D

File: lang/main.json [hunk 18]
```diff
@@ -773,6 +773,7 @@
         "enterRoomTitle": "Start a new meeting",
         "go": "GO",
         "join": "JOIN",
+        "numbers": "Info",
```
Comment (saghul) [inline lang/main.json]: maybe set the key to "info" as well?

File: react/features/base/react/components/native/HeaderNavigation.js [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+import HeaderLabel from './HeaderLabel';
+import styles from './headerstyles';
+
+
+type Props = {
+
+    /**
+     * Boolean to set the forward button disabled.
+     */
+    forwardDisabled: boolean,
```
Comment (saghul) [inline react/features/base/react/components/native/HeaderNavigation.js]: If the button is not shown when `onPressForward` is not passed, why would someone show a disabled button? And why wouldn't we allow the same for the back button?

File: react/features/base/react/components/native/HeaderNavigation.js [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * Callback to be invoked on pressing the forward button.
+     */
+    onPressForward: ?Function,
+}
+
+/**
+ * Implements the standard navigation content for the header.
+ */
+class HeaderNavigation extends Component<Props> {
```
Comment (saghul) [inline react/features/base/react/components/native/HeaderNavigation.js]: This could have gone in a separate commit.

File: react/features/chat/components/native/Chat.js [hunk 21]
```diff
@@ -42,8 +41,9 @@ class Chat extends AbstractChat<Props> {
                     behavior = 'padding'
                     style = { styles.chatContainer }>
                     <Header>
-                        <BackButton onPress = { this.props._onToggleChat } />
-                        <HeaderLabel labelKey = 'chat.title' />
+                        <HeaderNavigation
```
Comment (saghul) [inline react/features/chat/components/native/Chat.js]: Hum, this looks a bit odd. Wouldn't it mke sense to have a `HeaderWithNavigation` component which extends header adding the buttons, instead of needing to nest the components?

File: react/features/invite/actions.js [hunk 22]
```diff
@@ -268,3 +278,19 @@ export function removePendingInviteRequests() {
         type: REMOVE_PENDING_INVITE_REQUESTS
     };
 }
+
+/**
+ * Action to show the dial in summary.
```
Comment (saghul) [inline react/features/invite/actions.js]: show -> set

File: react/features/invite/components/add-people-dialog/native/AddPeopleDialog.js [hunk 23]
```diff
@@ -147,12 +145,12 @@ class AddPeopleDialog extends AbstractAddPeopleDialog<Props, State> {
                 onRequestClose = { this._onCloseAddPeopleDialog }
                 visible = { this.props._isVisible }>
                 <Header>
-                    <BackButton onPress = { this._onCloseAddPeopleDialog } />
-                    <HeaderLabel labelKey = 'inviteDialog.header' />
-                    <ForwardButton
-                        disabled = { this._isAddDisabled() }
-                        labelKey = 'inviteDialog.send'
-                        onPress = { this._onInvite } />
+                    <HeaderNavigation
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/native/AddPeopleDialog.js]: ditto (the HeaderWithNavigation thing)

File: react/features/base/react/components/native/HeaderNavigation.js (hunk 19 above)
Comment (zbettenbuk) [reply react/features/base/react/components/native/HeaderNavigation.js]: We don't need that possibility for the back button (there is no use-case) so I didn't implement that, trying to avoid to have a too generic component with unused functions, like the old Dialog was.

About the disabled thing: if a forward button is not rendered, then no disabled button will render. These props go together but I don't like complex props, so separated them. If there is no ```onPressForward``` then the ```forwardDisabled``` has no effect

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js [hunk 24]
```diff
@@ -0,0 +1,156 @@
...
+     */
+    _onNavigate(request) {
+        const { url } = request;
+
+        if (url.startsWith('tel:')) {
+            Linking.openURL(url);
+            this.props.dispatch(hideDialInSummary());
+        }
+
+        return url === getDialInfoPageURLForURIString(this.props._summaryUrl);
```
Comment (saghul) [inline react/features/invite/components/dial-in-summary/native/DialInSummary.js]: what does the return value do?

File: react/features/invite/components/dial-in-summary/native/DialInSummaryErrorDialog.js [hunk 25]
```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import React, { Component } from 'react';
+
+import { ErrorDialog } from '../../../../base/dialog';
+import { translate } from '../../../../base/i18n';
+import { connect } from '../../../../base/redux';
+
+/**
+ * Dialog to confirm a remote participant mute action.
```
Comment (saghul) [inline react/features/invite/components/dial-in-summary/native/DialInSummaryErrorDialog.js]: c&p error

File: react/features/invite/components/dial-in-summary/native/DialInSummaryErrorDialog.js [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+class DialInSummaryErrorDialog extends Component<{}> {
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <ErrorDialog
```
Comment (saghul) [inline react/features/invite/components/dial-in-summary/native/DialInSummaryErrorDialog.js]: I'm not sure I like this being called "ErrorDialog", as it has no relation with any error condition, except being called here. How about "AlertDialog"? Akin to the native UIAlertView (and whatever else on Android).

File: react/features/invite/components/dial-in-summary/native/DialInSummaryErrorDialog.js (hunk 25 above)
Comment (zbettenbuk) [reply react/features/invite/components/dial-in-summary/native/DialInSummaryErrorDialog.js]: Right :(

File: react/features/invite/components/dial-in-summary/native/DialInSummary.js (hunk 24 above)
Comment (zbettenbuk) [reply react/features/invite/components/dial-in-summary/native/DialInSummary.js]: This is to only let the webview navigate to the info page and nowhere else
